### PR TITLE
🐛 Use widest possible src from srcset for shrunk auto-lightbox images

### LINKED ISSUE #28304: lightbox-auto not created when using srcset
## What's the issue?
When using srcset in an amp-img tag, lightbox-auto isn't set-up.
All conditions for lightbox-auto to work have been met.

## What isn't working?
When using images with srcset, lightbox isn't autmatically created with lighbox-auto.

## What is the expected behavior?
Images defined with srcset should have a lightbox created (with lightbox-auto), same as images with only src

## How do we reproduce the issue?

Create an amp-img with srcset and one without, as done here: https://jsbin.com/mutecek/edit?html,output

## What browsers are affected?
Tested in 2 browsers. Both failing. I guess this will fail on all browsers.

Google Chrome: 81.0.4044.138 (Official Build) (64-bit)
Revision: 8c6c7ba89cc9453625af54f11fd83179e23450fa-refs/branch-heads/4044@{#999}
OS: Linux
JavaScript: V8 8.1.307.32

Firefox: 76.0 (64-bit)

## Which AMP version is affected?
AMP version: 2004252135000

### PR DESCRIPTION
**Pull request by alanorozco**
A lower resolution image can be selected from a `srcset`. Auto-lightbox uses the `.naturalWidth/Height` properties to determine if the image is shrunk in its render box from a higher resolution, and can therefore be expanded without loss.

Since the lightbox already knows how to use a higher-resolution src from `srcset`, auto-lightbox should consider the highest possible intrinsic width in the `srcset` to determine if there's a higher resolution image that can be used.

Fixes #28304

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
Are there any tests that could help to avoid breakages going forward?

File: extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js [hunk 1]
```diff
@@ -174,6 +175,57 @@ export class Criteria {
...
+ * Regex for the width-selection portion of a srcset, so for the
+ * general grammar: (URL [NUM[w|x]],)*, this should express "NUMw".
+ * E.g. in "image1.png 100w, image2.png 50w", this matches "100w" and "50w"
+ */
+const srcsetWidthRe = /\s+([0-9]+)w(,|$)/g;
+
+/**
+ * Parses srcset partially to get the maximum defined intrinsic width.
+ * @param {!Element} img
+ * @return {number} -1 if no srcset, or if srcset is defined by dpr instead of
```
Comment (kristoferbaxter) [inline extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js:187]: Clever using `-1` here as no `srcset` value in that case, but perhaps you could explain why (I'm assuming it's the `maxWidthFromSrcset > naturalWidth)` check below.

Comment (alanorozco) [reply extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js:187]: Good point!

File: extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js [hunk 2]
```diff
@@ -174,6 +174,57 @@ export class Criteria {
   }
 }
 
+/**
+ * Regex for the width-selection portion of a srcset, so for the
+ * general grammar: (URL [NUM[w|x]],)*, this should express "NUMw".
+ * E.g. in "image1.png 100w, image2.png 50w", this matches "100w" and "50w"
+ */
+const srcsetWidthRe = /\s+([0-9]+)w(,|$)/g;
```
Comment (wassgha) [inline extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js:182]: See if we have this anywhere else? Looks like it should be a util

**Comment by alanorozco** [comment]
@kristoferbaxter Added an e2e test.

File: extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js (hunk 2 above)
Comment (alanorozco) [reply extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js]: There's the other `parseSrcset` that will give you a fully parsed object.

This use case for getting only the width-selection values is very particular, I'm not convinced this util is useful elsewhere.

**Comment by alanorozco** [comment]
@wassgha Nothing to check on cache side.

**Pull request merged**